test/integration/spec/connect.js

```diff
@@ -720,6 +720,58 @@ describe('connect', function() {
     });
   });
 
+  describe.only('called with a Room name and', () => {
+    let sid;
+    let cancelablePromise;
+
+    before(async () => {
+      sid = await createRoom(randomName(), defaults.topology);
+      const options = Object.assign({ name: sid, tracks: [] }, defaults);
+      cancelablePromise = connect(getToken(randomName()), options);
+    });
+
+    after(() => completeRoom(sid));
+
+    it('should return a promise', async () => {
+      let finallyCalled = false;
+      await cancelablePromise.finally(() => {
+        finallyCalled = true;
```

Comment:
Instead of using booleans, how about using stubs or spies? Something like

```js
const onFinally = sinon.stub();
await cancelablePromise.finally(onFinally);
sinon.assert.calledOnce(onFinally);
```

Reply:
@charliesantos I noticed that we weren't using sinons in any of our integration tests so I wasn't sure if I should add it for just this which is why I went the boolean route. I can absolutely use a spy though, do you think that'd be okay?

Reply:
@PikaJoyce yes I think it's better. Check logger.js integration tests for reference.